sdk/core/core-http/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -84,8 +99,21 @@ export class BearerTokenAuthenticationPolicy extends BaseRequestPolicy {
   private async getToken(options: GetTokenOptions): Promise<string | undefined> {
     let accessToken = this.tokenCache.getCachedToken();
     if (accessToken === undefined) {
-      accessToken = (await this.credential.getToken(this.scopes, options)) || undefined;
-      this.tokenCache.setCachedToken(accessToken);
+      // Waiting for the next refresh only if the cache is unable to retrieve the access token,
+      // which means that it has expired, or it has never been set.
+      const refreshPromise = this.tokenRefresher.refresh(options);
+      if (refreshPromise !== null) {
+        accessToken = await refreshPromise;
+      }
```

Comment:
`refresh` no longer returns null, so I think we can do this:

```suggestion
      accessToken = await this.tokenRefresher.refresh(options);
```

Reply:
thank you, thank you 🌞